### DISCUSSION TITLE
Generic maildrop/tracking endpoints

### DISCUSSION THREAD

**Discussion by tobias-** [parcelvoy/platform#527] [Ideas]
I'm the primary developer of [tachikoma](https://github.com/SourceForgery/tachikoma), an ESP (email service provider) software similar to SES, Sendgrid, Mandrill et al.

Currently Parcelvoy supports a plethora of ESP's, but there is always going to be some that aren't supported, even rather big ones such as Brevo (previously SendInBlue) as it requires both a nodemailer module implementing the protocol and a change in the platform code to incorporate it.

My idea is that parcelvoy should implement generic maildrop and tracking endpoints that can be triggered by the other ESP or via a proxy that changes the incoming request before forwarding it, and such a proxy could in most cases be implemented with a couple of dozen lines of code that could be deployed on the lambda provider of choice. Exactly the format of the endpoints is of course up for discussion, but as long as the tracking endpoints keep accepting the previous format, any updates can be made, and similarly, adding extra data to the mail drop format, would allow the proxy to work.

Another idea is to have an advanced option to modify the endpoint the current providers go to, instead of having hardcoded url where they drop the email. That would basically work the same way, but would limit the amount of data that could be sent both ways to the subset supported by both Parcelvoy  and that supported ESP.

The company I worked for did provide an attempt for this, https://github.com/parcelvoy/platform/pull/467 but it was probably misunderstood.

**Comment by pushchris** [comment]
Not sure I'm understanding what you are actually proposing or how it would be helpful. Tracking (opens and clicks) is handled internally by the platform in a service agnostic way and pretty much every email provider out there allows for sending via SMTP. What is gained by this?
